### ISSUE TITLE
Create UTXO RPC query

### ISSUE THREAD

**Issue by btcdrak** [viacoin/viacoin-obsolete#3]
Query UTXO by address.

Something like similar to `listunspent` but for any arbitrary address

Returning a JSON string like insight-api returns for `/api/addr/<address>/utxo`

``` json
[
    {
        "txid" : "5eb995cc059740bc483fe9ce8dcbc2b16449010213c3dd7622d5dd6ecca3c0b3",
        "vout" : 1,
        "address" : "Vh9DsgqDfbobnpBWnqkF2wUfmGDccqocUA",
        "scriptPubKey" : "cf4d9b28bb976a9144e71ac4bfdd2fe3443097ab8f794e88ac",
        "amount" : 12.00000,
        "confirmations" : 47,
        "spendable" : true
    },
    {
        "txid" : "10213c3dd75eb995cc059740bc483fe9622d5dd6ecca3c0b3ce8dcbc2b164490",
        "vout" : 1,
        "address" : "VkF2wUfmh9bnpBDsDccqocUAgqDfboWnqG",
        "scriptPubKey" : "b28bb9443097ab8f794e88ac76a9144e71ac4bfdd2fe3cf4d9",
        "amount" : 156.99900000,
        "confirmations" : 10,
        "spendable" : true
    }
]
```

/cc @reorder

**Comment by reorder** [comment]
I suggest merging https://github.com/bitcoin/bitcoin/pull/3652 and extending listunspent to support arbitrary addresses. However, to fully match clearinghoused semantics an additional boolean parameter might be useful indicating whether to include mempool transactions in the output.

**Comment by btcdrak** [comment]
@reorder I guess you mean `confirmations = 0`. All bitcoin RPC calls include minimum confirmations (`minconf`) as an argument. It is essential that utxo list 0 confirmations if minconf=0 (just as listunspent does - it has args `<minconf> <maxconf> <address>`).

**Comment by reorder** [comment]
@btcdrak I was refering to `gettxout` call actually which is basically what we need but working with transations, not addresses. It has  `includemempool` boolean parameter. If the new call is implemented like `listunspent` or  `listunspent` is modified to support arbitrary addresses, then `minconf=0` will work indeed.

**Comment by btcdrak** [comment]
@reorder we'd still need an addressindex.

**Comment by reorder** [comment]
@btcdrak Of course it is still needed. bitcoin/bitcoin/pull/3652 alone would actually suffice, but extending listunspent seems attractive wrt usability.

**Comment by btcdrak** [comment]
I'm tempted to merge this into `0.10` from `utxo_aware`

**Comment by btcdrak** [comment]
So I've merged in `utxo_aware` to `0.10` so it will be easy to run clearwallet federated nodes - it's not in `master` for the time being (as there are open PRs upstream).

**Comment by btcdrak** [comment]
`addrindex` was merged to master.
